hadoop/src/main/java/org/apache/carbondata/hadoop/util/CarbonInputFormatUtil.java

```diff
@@ -70,6 +69,18 @@ public static CarbonQueryPlan createQueryPlan(CarbonTable carbonTable, String co
           i++;
         }
       }
+    } else {
+      // query all columns
+      List<CarbonMeasure> tableMsrs = carbonTable.getMeasureByTableName(factTableName);
```

Comment:
ok, I got the issue here. Can we do like as following
1. if it is come from count(*), we set the projection column to a special string (like empty string "") in configuration, then we can process it differently in `createQueryPlan`. 
2. If user want to project all column, projection column should be set to null or list all columns.
3. If user want to project specific columns, then set it to specific columns